design-system/src/theme-provider.visualroute.jsx

```diff
@@ -1,45 +1,93 @@
-import { ThemeProvider, useTheme, customProperties } from '@commercetools-uikit/design-system';
-import { Suite, Spec } from '../../test/percy';
+import { useTheme, customProperties } from '@commercetools-uikit/design-system';
+import kebabCase from 'lodash/kebabCase';
+import PropTypes from 'prop-types';
+import {
+  Suite,
+  Spec,
+  LocalDarkThemeProvider,
+  LocalThemeProvider,
+} from '../../test/percy';
 
 export const routePath = '/theme-provider';
 
-const DummyComponent = () => {
+const DummyComponent = (props) => {
   const { theme } = useTheme();
   return (
     <h1
       style={{
-        color: customProperties.colorPrimary,
+        color: props.color
+          ? `var(--${kebabCase(props.color)})`
+          : customProperties.colorSolid,
+        backgroundColor: customProperties.colorSurface,
+        margin: 0,
       }}
     >
-      Title with {theme} theme <i>colorPrimary</i> design token
+      {props.title ?? (
+        <>
+          Title with {theme} theme <i>colorSolid</i> design token
+        </>
+      )}
     </h1>
   );
 };
+DummyComponent.propTypes = {
+  color: PropTypes.string,
+  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+};
 
 export const component = () => (
   <Suite>
     <Spec label="use default theme">
-      <ThemeProvider scope="local">
+      <LocalThemeProvider>
         <DummyComponent />
-      </ThemeProvider>
+      </LocalThemeProvider>
     </Spec>
 
     <Spec label="use dark theme">
-      <ThemeProvider scope="local" theme="dark">
+      <LocalDarkThemeProvider theme="dark">
         <DummyComponent />
-      </ThemeProvider>
+      </LocalDarkThemeProvider>
     </Spec>
 
     <Spec label="repeat default theme">
-      <ThemeProvider scope="local">
+      <LocalThemeProvider>
         <DummyComponent />
-      </ThemeProvider>
+      </LocalThemeProvider>
     </Spec>
 
     <Spec label="repeat dark theme">
-      <ThemeProvider scope="local" theme="dark">
+      <LocalDarkThemeProvider theme="dark">
         <DummyComponent />
-      </ThemeProvider>
+      </LocalDarkThemeProvider>
+    </Spec>
+
+    <Spec label="overridden default theme">
+      <LocalThemeProvider customPropertiesOverrides={{ colorSolid: 'red' }}>
+        <DummyComponent
+          title={
+            <>
+              Title with overridden <i>colorSolid</i> design token
+            </>
+          }
+        />
+      </LocalThemeProvider>
+    </Spec>
+
+    <Spec label="custom property with double hyphen and kebab-case naming added to default theme">
```

Comment:
Sorry, but I don't understand what we are testing here (also in the next test).
Could you maybe explain a little further:

Reply:
Sure, the purpose was to override the theme with custom properties that are not part of standard `customProperties` exposed from `@commercetools-uikit/design-system` and test the theme provider and a component in that case.

Also, to make users aware that they can use `css variables` notation directly and camel cased naming (for anyone using tests as guidance how to use the feature).

I don't think this test is crucial, perhaps it belongs to unit tests of the `<ThemeProvider>`